matrix/MyMatrix.py

```diff
@@ -1,22 +1,114 @@
+                foo -= (self.matrix[i][j]*self.results[j])
+            self.results[i] = foo / self.matrix[i][i]
+
+    def gaussPartial(self):
+        foo = [0 for i in range(0, self.size)]
+        pointers = [i for i in range(0, self.size)]
+        for i in range(0, self.size):
+            for j in range(0, self.size):
+                temp = [foo[i], abs(self.matrix[i][j])]
+                foo[i] = max(temp)
+        for i in range(0, self.size-1):
+            maxValue = 0
+            for j in range(i, self.size):
+                y = abs(self.matrix[pointers[j]][i]/foo[pointers[j]])
+                if y > maxValue:
+                    maxValue = y
+                    t = j
+            temp = pointers[i]
+            pointers[i] = pointers[t]
+            pointers[t] = temp
+            for j in range (i+1, self.size):
+                self.matrix[pointers[j]][i] = self.matrix[pointers[j]][i] / self.matrix[pointers[i]][i]
+                for k in range(i+1, self.size):
+                    self.matrix[pointers[j]][k] -= (self.matrix[pointers[j]][i]*self.matrix[pointers[i]][k])
+        for i in range(0, self.size-1):
+            for j in range(0, self.size):
+                self.vector[pointers[j]] -= (self.matrix[pointers[j]][i]*self.vector[pointers[i]])
+        for i in range(self.size-1, -1, -1):
+            foo = self.vector[pointers[i]]
+            for j in range(i+1, self.size):
+                foo -= (self.matrix[pointers[i]][j]*self.results[j])
+            self.results[i] = foo / self.matrix[pointers[i]][i]
+
+
+    def gaussComplete(self):
+        columns = [i for i in range (0, self.size)]
+        rows = [i for i in range (0, self.size)]
+        for i in range(0, self.size):
+            maxValue = 0
+            for j in range(i, self.size):
+                for k in range(i, self.size):
+                    y = abs(self.matrix[rows[j]][columns[k]])
+                    if y > maxValue:
+                        maxValue = y
+                        rowIndex = j
+                        columnIndex = k
+            self.swapRows(i, rowIndex)
+            self.swapColumns(i, columnIndex)
+            for j in range(i+1, self.size):
+                self.matrix[rows[j]][columns[i]] = self.matrix[rows[j]][columns[i]] / self.matrix[rows[i]][columns[i]]
+                for k in range (i+1, self.size):
+                    self.matrix[rows[j]][columns[k]] -= (self.matrix[rows[j]][columns[i]] * self.matrix[rows[i]][columns[k]])
+
+        for i in range(0, self.size-1):
+            for j in range(i+1, self.size):
+                self.vector[rows[j]] -= (self.matrix[rows[j]][columns[i]] * self.vector[rows[i]])
+
+        for i in range(self.size-1, -1, -1):
+            foo = self.vector[rows[i]]
+            for j in range(i+1, self.size):
+                foo -= (self.matrix[rows[i]][columns[j]] * self.results[columns[j]])
+            self.results[columns[i]] = foo / self.matrix[rows[i]][columns[i]]
```

Comment:
Na pewno wszędzie tutaj trzeba będzie popoprawiać działania na liczbach
Raczej później, nie wiem czy dziś, spojrzę się, czy to wszystko ma sens samo w sobie, czy działa w ogóle